pkg/adapter/v2/cloudevents.go

```diff
@@ -0,0 +1,144 @@
+/*
+Copyright 2020 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package adapter
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	cloudevents "github.com/cloudevents/sdk-go/v2"
+	"github.com/cloudevents/sdk-go/v2/event"
+	"github.com/cloudevents/sdk-go/v2/protocol/http"
+	duckv1 "knative.dev/pkg/apis/duck/v1"
+	"knative.dev/pkg/source"
+)
+
+// NewCloudEventsClient returns a client that will apply the ceOverrides to
+// outbound events and report outbound event counts.
+func NewCloudEventsClient(target string, ceOverrides *duckv1.CloudEventOverrides, reporter source.StatsReporter) (cloudevents.Client, error) {
```

Comment:
Should this live in kncloudevents?

Reply:
I think so. I'll have soon a PR doing this (and a bit more)